extensions/parquet/base/src/main/java/io/deephaven/parquet/base/ParquetFileReader.java

```diff
@@ -27,27 +29,53 @@ public class ParquetFileReader {
     private static final int FOOTER_LENGTH_SIZE = 4;
     private static final String MAGIC_STR = "PAR1";
     static final byte[] MAGIC = MAGIC_STR.getBytes(StandardCharsets.US_ASCII);
+    public static final String S3_PARQUET_FILE_URI_SCHEME = "s3";
+    private static final String S3_PARQUET_FILE_URI_PREFIX = "s3://";
 
     public final FileMetaData fileMetaData;
     private final SeekableChannelsProvider channelsProvider;
     private final Path rootPath;
     private final MessageType type;
 
+    // TODO Where should I keep it?
+    public static URI convertToURI(final String filePath) {
+        // We need to encode spaces in the path string that are not allowed in a URI
+        if (filePath.startsWith(S3_PARQUET_FILE_URI_PREFIX)) {
+            try {
+                return new URI(filePath.replace(" ", "%20"));
```

Comment:
Here I am replacing spaces in the names with %20.
The problem is that I need to AWS URI encode this file path. This can be done using a aws client object which is present in the S3SeekableByteChannel. 
Link: https://aws.amazon.com/blogs/devops/s3-uri-parsing-is-now-available-in-aws-sdk-for-java-2-x/

This is the initialization part and I don't want to complicate it too much because till this point, the client object is not even created. So for now, I am just handling spaces as special characters in the file name.

Reply:
I don't think we be trying to parse S3 URLs into S3 URIs, at least not at this layer. I expect URI.create should suffice?

```
jshell> URI.create("file:///tmp/my-file.parquet")
$7 ==> file:///tmp/my-file.parquet

jshell> URI.create("/tmp/my-file.parquet")
$8 ==> /tmp/my-file.parquet

jshell> URI.create("s3://my-bucket/path/to/my-file.parquet")
$9 ==> s3://my-bucket/path/to/my-file.parquet
```